src/overseer/store/jdbc.clj

```diff
@@ -0,0 +1,238 @@
+        (std/map-kv (partial map second) (group-by first edges))
+
+        job->dep-maps
+        (fn [job]
+          (when (contains? dep-edges job)
+            (->> (get dep-edges job)
+                 (map (fn [dep]
+                        {:job_id (:job/id job)
+                         :dep_id (:job/id dep)})))))]
+    (->> nodes
+         (mapcat job->dep-maps)
+         (filter identity))))
+
+(def default-table-opts
+  "Backend-specific system configuration options"
+  {:mysql
+   {:table-spec "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=DYNAMIC"}})
+
+(defn install' [adapter db-spec]
+  (j/db-do-commands
+    db-spec
+    [(j/create-table-ddl
+       :overseer_jobs
+       [[:id "VARCHAR(64) PRIMARY KEY"]
+        [:type "VARCHAR(255)"]
+        [:args "VARCHAR(2048)"]
+        [:status "VARCHAR(255)"]
+        [:failure "VARCHAR(2048)"]
+        [:heartbeat :int]
+        [:lock_version "INT NOT NULL DEFAULT '0'"]]
+       (get default-table-opts adapter {}))
+     (j/create-table-ddl
+       :overseer_dependencies
+       [[:job_id "VARCHAR(64)"]
+        [:dep_id "VARCHAR(64)"]]
+       (get default-table-opts adapter {}))
+     "CREATE INDEX index_overseer_jobs_on_status ON overseer_jobs (status)"
+     "CREATE INDEX index_overseer_dependencies_on_job_id ON overseer_dependencies (job_id)"
+     "CREATE INDEX index_overseer_dependencies_on_dep_id ON overseer_dependencies (dep_id)"])
+  :ok)
+
+(defn- jobs-with-unfinished-dep
+  "Return the seq subset of `job-ids` that have an unfinished direct dependency"
+  [db-spec job-ids]
+  (if (seq job-ids)
+    (->> (j/query
+           db-spec
+           (flatten [(str "SELECT job_id FROM overseer_dependencies "
+                          "INNER JOIN overseer_jobs ON overseer_jobs.id = overseer_dependencies.dep_id "
+                          (format "WHERE overseer_dependencies.job_id IN (%s) " (in-placeholders job-ids))
+                          "AND overseer_jobs.status != 'finished'")
+                     job-ids]))
+         (map :job_id))
+    []))
+
+(defmacro with-ignore-stale-objects [& body]
+  `(try
+    (do ~@body)
+    (catch clojure.lang.ExceptionInfo ex#
+      (when-not (= :stale-object (:type (ex-data ex#)))
+        (throw ex#)))))
+
+(defrecord JdbcStore [adapter db-spec]
+  core/Store
+  (install [this]
+    (install' adapter db-spec))
+
+  (transact-graph [this graph]
+    (assert (core/valid-graph? graph))
+    (let [job-rows (map job->jdbc-map (loom.graph/nodes graph))
+          dep-rows (job-dep-jdbc-maps graph)]
+      (j/with-db-transaction [t-con db-spec]
+        (j/insert-multi! t-con :overseer_jobs job-rows)
+        (j/insert-multi! t-con :overseer_dependencies dep-rows))))
+
+  (job-info [this job-id]
+    (query-job db-spec job-id))
+
+  (reserve-job [this job-id]
+    (let [where-map {:status "unstarted"}
+          set-map {:status "started"
+                   :heartbeat (core/heartbeat)}]
```

Comment:
I'm not clear if `(core/heartbeat)` existed before (it looks like it did?) but in single-write-master systems like Datomic or MySQL you can have the database itself assign timestamps or monotonically increasing IDs, which guarantees a single timeline / that there is no weird clock skew across nodes.  Under this system, if the works have misaligned clocks, stuff will get weird, which is bad.